--- target/assembly/bin/inc	2020-10-27 09:19:51.940334549 +0100
+++ target/assembly/bin/inc	2020-10-27 09:36:25.438005471 +0100

@@ -281,6 +281,9 @@ setupVendorSepcifics() {

 setupDefaults() {
     DEFAULT_JAVA_OPTS="-XX:+UnlockDiagnosticVMOptions "
+    DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:+HeapDumpOnOutOfMemoryError"
+    DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:+DisableExplicitGC"
+

     if [ "${VERSION}" -gt "11" ]; then
       # TODO revisit EventAdminImpl to avoid use of Subject.getSubject(AccessController.getContext());
@@ -292,6 +295,13 @@ setupDefaults() {

     DEFAULT_JAVA_OPTS="${DEFAULT_JAVA_OPTS} ${JAVA_VENDOR_OPTS}"

+    # Add default security file option
+    if [ "x$ODL_JAVA_SECURITY_PROPERTIES" != "x" ]; then
+        DEFAULT_JAVA_OPTS="-Djava.security.properties="${ODL_JAVA_SECURITY_PROPERTIES}" $DEFAULT_JAVA_OPTS"
+    else
+        DEFAULT_JAVA_OPTS="-Djava.security.properties="${KARAF_ETC}/odl.java.security" $DEFAULT_JAVA_OPTS"
+    fi
+
     DEFAULT_JAVA_DEBUG_PORT="5005"
     if [ "x${JAVA_DEBUG_PORT}" = "x" ]; then
         JAVA_DEBUG_PORT="${DEFAULT_JAVA_DEBUG_PORT}"
