--- target/assembly/bin/inc	2020-10-27 09:19:51.940334549 +0100
+++ target/assembly/bin/inc	2020-10-27 09:36:25.438005471 +0100
@@ -190,7 +190,7 @@
     fi

     if [ "x${JAVA_HOME}" = "x" ] && [ "${darwin}" = "true" ]; then
-        JAVA_HOME="`/usr/libexec/java_home -v 1.8`"
+        JAVA_HOME="`/usr/libexec/java_home -v 11`"
     fi
     if [ "x${JAVA_HOME}" = "x" ] && [ -r /etc/gentoo-release ] ; then
         JAVA_HOME=`java-config --jre-home`
@@ -243,9 +243,9 @@

     VERSION=`"${JAVA}" -version 2>&1 | ${AWK} -F '"' '/version/ {print $2}' | sed -e 's/_.*//g; s/^1\.//g; s/\..*//g; s/-.*//g;'`

-    # java must be at least version 8
-    if [ "${VERSION}" -lt "8" ]; then
-        die "JVM must be version 1.8 or greater"
+    # java must be at least version 11
+    if [ "${VERSION}" -lt "11" ]; then
+        die "JVM must be version 11 or greater"
     fi
 }

@@ -281,11 +281,20 @@
 setupDefaults() {

     DEFAULT_JAVA_OPTS="-XX:+UnlockDiagnosticVMOptions "
+    DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:+HeapDumpOnOutOfMemoryError"
+    DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:+DisableExplicitGC"

     setupVendorSepcifics

     DEFAULT_JAVA_OPTS="${DEFAULT_JAVA_OPTS} ${JAVA_VENDOR_OPTS}"

+    # Add default security file option
+    if [ "x$ODL_JAVA_SECURITY_PROPERTIES" != "x" ]; then
+        DEFAULT_JAVA_OPTS="-Djava.security.properties="${ODL_JAVA_SECURITY_PROPERTIES}" $DEFAULT_JAVA_OPTS"
+    else
+        DEFAULT_JAVA_OPTS="-Djava.security.properties="${KARAF_ETC}/odl.java.security" $DEFAULT_JAVA_OPTS"
+    fi
+
     DEFAULT_JAVA_DEBUG_PORT="5005"
     if [ "x${JAVA_DEBUG_PORT}" = "x" ]; then
         JAVA_DEBUG_PORT="${DEFAULT_JAVA_DEBUG_PORT}"
