--- target/assembly/bin/inc.orig	2018-08-21 16:15:40.146423070 +0200
+++ target/assembly/bin/inc	2018-08-21 16:15:43.798408830 +0200
@@ -257,6 +257,8 @@
     fi
 
     DEFAULT_JAVA_OPTS="-Xms${JAVA_MIN_MEM} -Xmx${JAVA_MAX_MEM} -XX:+UnlockDiagnosticVMOptions "
+    DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:+HeapDumpOnOutOfMemoryError"
+    DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:+DisableExplicitGC"
 
     #Set the JVM_VENDOR specific JVM flags
     if [ "${JVM_VENDOR}" = "SUN" ]; then
@@ -285,6 +287,13 @@
         fi
     fi
 
+    # Add default security file option
+    if [ "x$ODL_JAVA_SECURITY_PROPERTIES" != "x" ]; then
+        DEFAULT_JAVA_OPTS="-Djava.security.properties="${ODL_JAVA_SECURITY_PROPERTIES}" $DEFAULT_JAVA_OPTS"
+    else
+        DEFAULT_JAVA_OPTS="-Djava.security.properties="${KARAF_ETC}/odl.java.security" $DEFAULT_JAVA_OPTS"
+    fi
+
     DEFAULT_JAVA_DEBUG_PORT="5005"
     if [ "x${JAVA_DEBUG_PORT}" = "x" ]; then
         JAVA_DEBUG_PORT="${DEFAULT_JAVA_DEBUG_PORT}"
