--- target/assembly/bin/karaf	2017-09-19 11:52:37.274203647 +0200
+++ target/assembly/bin/karaf	2017-09-19 11:54:28.482544108 +0200
@@ -169,6 +169,13 @@
         KARAF_EXEC=""
     fi
 
+    # Use /dev/urandom to avoid blocking on /dev/random
+    # See http://www.2uo.de/myths-about-urandom/ to understand why this is safe (as long as your VM provisioning seeds
+    # the PRNG)
+    # The /dev/./urandom workaround is necessary because of https://bugs.openjdk.java.net/browse/JDK-6202721
+    NON_BLOCKING_PRNG=
+    [ -c /dev/urandom -a -r /dev/urandom ] && NON_BLOCKING_PRNG=-Djava.security.egd=file:/dev/./urandom
+
     debug=false
     nodebug=false
     while [ "${1}" != "" ]; do
@@ -256,8 +263,12 @@
         fi
 
         if [ "${ROOT_INSTANCE_RUNNING}" = "false" ] || [ "${CHECK_ROOT_INSTANCE_RUNNING}" = "false" ] ; then
+            # Using command line arguments as java arguments
+            # (as opposed to arguments for $MAIN)
+            # FIXME: document this in the user guide
             if [ "${VERSION}" -gt "80" ]; then
                 ${KARAF_EXEC} "${JAVA}" ${JAVA_OPTS} \
+                    "$NON_BLOCKING_PRNG" \
                     --add-opens java.base/java.security=ALL-UNNAMED \
                     --add-opens java.base/java.net=ALL-UNNAMED \
                     --add-opens java.base/java.lang=ALL-UNNAMED \
@@ -280,10 +291,12 @@
                     ${KARAF_SYSTEM_OPTS} \
                     ${KARAF_OPTS} \
                     ${OPTS} \
+                    "$@" \
                     -classpath "${CLASSPATH}" \
-                    ${MAIN} "$@"
+                    ${MAIN}
             else
                 ${KARAF_EXEC} "${JAVA}" ${JAVA_OPTS} \
+                    "$NON_BLOCKING_PRNG" \
                     -Djava.endorsed.dirs="${JAVA_ENDORSED_DIRS}" \
                     -Djava.ext.dirs="${JAVA_EXT_DIRS}" \
                     -Dkaraf.instances="${KARAF_HOME}/instances" \
@@ -297,8 +310,9 @@
                     ${KARAF_SYSTEM_OPTS} \
                     ${KARAF_OPTS} \
                     ${OPTS} \
+                    "$@" \
                     -classpath "${CLASSPATH}" \
-                    ${MAIN} "$@"
+                    ${MAIN}
             fi
         else
             die "There is a Root instance already running with name ${ROOT_INSTANCE_NAME} and pid ${ROOT_INSTANCE_PID}. If you know what you are doing and want to force the run anyway, export CHECK_ROOT_INSTANCE_RUNNING=false and re run the command."
